client/src/commands/code_analysis.ts

```diff
@@ -119,6 +119,42 @@ let resultsToDiagnostics = (
           }
         }
       }
+
+      if (item.message.endsWith(" is never used")) {
```

Comment:
Staying safe here and only targeting this message since these items seem to be the only ones you can just straight up remove via the reported range, without risking breaking the syntax (like if you'd remove a single variant constructor but not `|`, etc).